src/sorcha/ephemeris/simulation_setup.py

```diff
@@ -104,3 +116,80 @@ def generate_simulations(ephem, gm_sun, orbits_df):
         sim_dict[row["ObjID"]]["ex"] = ex
 
     return sim_dict
+
+
+def precompute_pointing_information(pointings_df, args, configs):
```

Comment:
This does the precomputation work. There are two instances of `pandas.apply` being used in here. Both of them are a result of not being able to _easily_ vectorize the code. 

However, it should be noted that with some additional time, I do think that we could vectorize those two calls and get a little more speed out of this.

Reply:
In addition, there are a few `for` loops over the rows (in the form of `enumerate`), but I don't think it is worth continuing to optimize them unless they turn into a bottleneck.